src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -36,7 +36,7 @@ public static int IndexOf(ref byte searchSpace, int searchSpaceLength, ref byte
             nuint valueTailNLength = (nuint)(uint)valueTailLength;
 
             // Avx2 implies Sse2
-            if (Sse2.IsSupported && searchSpaceLength - valueTailLength >= Vector128<byte>.Count)
+            if ((Sse2.IsSupported || AdvSimd.IsSupported) && searchSpaceLength - valueTailLength >= Vector128<byte>.Count)
```

Comment:
`Vector128.IsHardwareAccelerated`?

Reply:
@tannergooding is `Vector128.IsHardwareAccelerated` the same as Sse2? it's not clear what it returns 
Same question about `Vector256.IsHardwareAccelerated` - is it Avx1 or Avx2 ?

Reply:
It matches the underlying ISA "implications" here: https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt#L78-L79

So technically speaking `Sse` and `Avx`; and then `AdvSimd` for ARM64.

For `Vector128`, this "doesn't matter", realistically. `Sse2` is considered baseline and outside of testing scenarios users won't be disabling `Sse2` and leaving on `Sse`.

For `Vector256`, I see how this might be problematic and so we should probably specially handle `get_IsHardwareAccelerated` in https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsic.cpp#L288